lms/djangoapps/teams/tests/test_views.py

```diff
@@ -523,3 +568,224 @@ def test_invalid_course_id(self, course_id):
 
     def test_invalid_topic_id(self):
         self.get_topic_detail('foobar', self.test_course_1.id, 404)
+
+
+@ddt.ddt
+class TestListMembershipAPI(TeamAPITestCase):
+    """Test cases for the membership list endpoint."""
+
+    @ddt.data(
+        (None, 403),
+        ('student_inactive', 403),
+        ('student_unenrolled', 404),
+        ('student_enrolled', 200),
+        ('student_enrolled_both_courses_other_team', 200),
+        ('staff', 200),
+        ('course_staff', 200),
+    )
+    @ddt.unpack
+    def test_access(self, user, status):
+        membership = self.get_membership_list(status, {'team_id': self.test_team_1.team_id}, user=user)
+        if status == 200:
+            self.assertEqual(membership['count'], 1)
+            self.assertEqual(membership['results'][0]['user']['id'], self.users['student_enrolled'].username)
+
+    @ddt.data(
+        (None, 403, False),
+        ('student_inactive', 403, False),
+        ('student_unenrolled', 200, False),
+        ('student_enrolled', 200, True),
+        ('student_enrolled_both_courses_other_team', 200, True),
+        ('staff', 200, True),
+        ('course_staff', 200, True),
+    )
+    @ddt.unpack
+    def test_access_by_username(self, user, status, has_content):
+        membership = self.get_membership_list(status, {'username': self.users['student_enrolled'].username}, user=user)
+        if status == 200:
+            if has_content:
+                self.assertEqual(membership['count'], 1)
+                self.assertEqual(membership['results'][0]['team']['id'], self.test_team_1.team_id)
+            else:
+                self.assertEqual(membership['count'], 0)
+
+    def test_no_username_or_team_id(self):
+        self.get_membership_list(400, {})
+
+    def test_bad_team_id(self):
+        self.get_membership_list(404, {'team_id': 'foobar'})
+
+    def test_expand_user(self):
+        result = self.get_membership_list(200, {'team_id': self.test_team_1.team_id, 'expand': 'user'})
+        self.assertIn('email', result['results'][0]['user'])
+
+    def test_expand_team(self):
+        result = self.get_membership_list(200, {'team_id': self.test_team_1.team_id, 'expand': 'team'})
+        self.assertIn('description', result['results'][0]['team'])
```

Comment:
It uses the exact same serializer as the team detail endpoint. id, name, is_active, course_id, topic_id, date_created, description, country, language, and membership are returned.